src/main/java/com/criticalsoftware/AnnouncementResource.java

```diff
@@ -0,0 +1,84 @@
+package com.criticalsoftware;
+
+import jakarta.inject.Inject;
+import jakarta.ws.rs.DELETE;
+import jakarta.ws.rs.PUT;
+import jakarta.ws.rs.Path;
+import jakarta.ws.rs.PathParam;
+import jakarta.ws.rs.core.Response;
+import org.bson.types.ObjectId;
+
+@Path("/announcements")
+public class AnnouncementResource {
+
+    @Inject
+    private AnnouncementRepository repository;
+
+    @DELETE
+    @Path("/{id}")
+    public Response delete(@PathParam("id") String id) {
+        try {
+            // Validate ObjectId
+            if (!ObjectId.isValid(id)) {
+                return Response.status(Response.Status.BAD_REQUEST).entity("Invalid announcement ID").build();
+            }
```

Comment:
I would say "Invalid Id format". What do you think?

Reply:
Done!